src/core/meshcop/border_agent.cpp

```diff
@@ -317,6 +337,217 @@ template <> void BorderAgent::HandleTmf<kUriRelayRx>(Coap::Message &aMessage, co
     FreeMessageOnError(message, error);
 }
 
+void BorderAgent::NotifyMeshCoPServiceChange(void)
+{
+    // We know all the TXT entries from OT core and the total size is less than 128.
+    static constexpr uint16_t kMaxTxtDataLen = 128;
+
+    uint8_t  txtDataBuffer[kMaxTxtDataLen];
+    uint16_t txtDataLen = kMaxTxtDataLen;
+
+    VerifyOrExit(mMeshCoPServiceChangedCallback.IsSet());
+    SuccessOrExit(mMeshCopTxtEncoder.EncodeTxtData(txtDataBuffer, txtDataLen));
```

Comment:
We are already assuming large enough buffer for this. 

But then in `EncodeTxtData()` we check for failure and return `error` and then skip here. Should we maybe assert instead?

Reply:
Make sense. Updated to `OT_ASSERT` here.